### ISSUE TITLE
Consistently apply and preserve source information in PUT handling 

### ISSUE THREAD

**Issue by preeve9534** [SignalK/signalk-server#1218]
I have a use case described briefly below which at the moment requires messy work-arounds because of what might be partial /inconsistent handling of source data in PUT request handling.

Use case. I want to raise PUT requests on electrical.switches.bank paths associated with an N2K relay output channels. User initiated PUT requests to operate a relay can come from either a plugin (which monitors a physical switch) or an app (which provides a virtual interface). I want my put handler to moderate its action based upon the source of an incoming PUT request.

__My current understanding__

1. I can use app.putSelf() to raise a PUT request from my plugin and I can specify a source property. Good.

2. I want to use a delta PUT request my app, but this appears not to support use of a source property. The specification example shows: 
```
{
  "context": "vessels.urn:mrn:signalk:uuid:6b0e776f-811a-4b35-980e-b93405371bc5",
  "requestId": "6b0e776f-811a-4b35-980e-b93405371bc5",
  "put": [{
	  "path": "a.b.c",
	  "value": 1234
  }]
}
```
I note that the HTTP example adjacent to this in the specification does allow a source property to be specified, so maybe the delta does too and it simply isn't documented in the spec.

3. The situation at (2) is, in any case, immaterial because the source property value is never passed to a PUT handler. The put handler signature is specified as:
```
f(context, path, value, callback)
```

My feeling is that the source property value may be a useful conditioner of put handler logic and should be available ... maybe as an optional fifth parameter?

**Comment by norbert-walter** [comment]
It is possible from outside to send PUT requests to SignalK server? I try with curl but  I cold not saving the data.

`curl -X PUT -H "Content-Type: application/json" -d '{"updates":[{"source":{"sentence":"MTW","talker":"WI","type":"NMEA0183"},"timestamp":"2021-01-26T11:27:33.896Z","values":[{"path":"sensors.engine_3.test","value":15.099999999999994}]}],"context":"vessels.self"}' http://192.168.1.125:3000`

What is wrong? I need a authentication? In compare with the documentation I could not see the difference. I try also to authenticate. Always coming the answare "Cannot POST / Cannot PUT"

https://signalk.org/specification/1.5.0/doc/put.html

**Comment by sbender9** [comment]
No, this is not how PUT is intended to be used. There's more clarification in this yet to be published doc: https://github.com/SignalK/specification/blob/put-clarification/gitbook-docs/put.md

**Comment by norbert-walter** [comment]
Ok, then is the document old and this way is not more possible, right?

**Comment by sbender9** [comment]
That way was never possible. The old version of the document does not say that you can do that. 

Yes, you need authentication. 

See the security document in the spec.

**Comment by sbender9** [comment]
In the future, please open a new issue instead of piggy backing on someone else's issue. 

Or go to https://github.com/SignalK/signalk/discussions if you need help with something.
